main.c

```diff
@@ -73,10 +73,38 @@ enum display_mode {
    DISPLAY_MODE_KHR,
 };
 
+typedef struct {
+   const char *name;
+   uint64_t value;
+} DrmModifierEntry;
+
+static const DrmModifierEntry drm_modifiers[] = {
+   { "DRM_FORMAT_MOD_LINEAR", DRM_FORMAT_MOD_LINEAR },
+   { "I915_FORMAT_MOD_Y_TILED_CCS", I915_FORMAT_MOD_Y_TILED_CCS },
+   { "I915_FORMAT_MOD_Yf_TILED_CCS", I915_FORMAT_MOD_Yf_TILED_CCS },
+   { "I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS", I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS },
+   { "I915_FORMAT_MOD_Y_TILED_GEN12_MC_CCS", I915_FORMAT_MOD_Y_TILED_GEN12_MC_CCS },
+   { "I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS_CC", I915_FORMAT_MOD_Y_TILED_GEN12_RC_CCS_CC },
+};
+
+#define DRM_MODIFIER_COUNT (sizeof(drm_modifiers) / sizeof(drm_modifiers[0]))
+static uint64_t drm_modifier = DRM_FORMAT_MOD_LINEAR;
 static enum display_mode display_mode = DISPLAY_MODE_AUTO;
 static uint32_t width = 1024, height = 768;
 static const char *arg_out_file = "./cube.png";
 static bool require_protected = false;
+static bool is_linear = true;
+
+static uint64_t lookup_drm_modifier(const char *name) {
```

Comment:
This should accept raw hex values too. If `name` starts with `0x` or `0X`, it should `sscanf` the modifier value.